### ISSUE TITLE
Game randomly crashes

### ISSUE THREAD

**Issue by girlbossdev** [fabiangreffrath/crispy-doom#1066]
### Background

Version of Crispy Doom: 6.0.0

Operating System and version: Windows 11 Pro 22H2 (build 22621.1778)

Game: Doom 2, TNT: Evilution

Any loaded WADs and mods (please include full command line): None other than the IWAD

### Bug description

Observed behavior: The game continues to run correctly

Expected behavior: The game closes without any error message after a seemingly random period of time, doesn't seem to be triggered by anything in particular

**Comment by QuartzSTQ** [comment]
I'm experiencing this as well, and I was just about to open an issue of my own. I can say that it's not limited to Doom 2 or TNT, it occurs in Doom 1 as well.

**Comment by rfomin** [comment]
@QuartzSTQ Are you using Windows 11 too?

Does Crispy Doom 5.12 work? Or 5.11?

**Comment by QuartzSTQ** [comment]
> @QuartzSTQ Are you using Windows 11 too?

Yes.

> Does Crispy Doom 5.12 work? Or 5.11?

Yes, and I've tried using the [autobuilds](http://latest.chocolate-doom.org/) as well, but they have the same issue. I thought it might've had something to do with the fact that I was using the 64-bit builds specifically, but I think I can rule that out, even though I haven't actually tried using the 32-bit builds of 6.0.0 (5.12 worked using the 64-bit build).

**Comment by rfomin** [comment]
Have you tested in a new clean directory? Also have you changed the default settings (eg different music settings)? Can you post your configs?

**Comment by rfomin** [comment]
> Have you tested in a new clean directory? Also have you changed the default settings (eg different music settings)? Can you post your configs (default.cfg and crispy-doom.cfg)?

**Comment by mikeday0** [comment]
You wouldn't happen to be using FLAC music replacements would you? There is an issue with that in the latest SDL Mixer that causes crashing whenever the song loops.

**Comment by girlbossdev** [comment]
I guess I'll close this since someone else already brought it up: #1049

**Comment by mikeday0** [comment]
FWIW the bug has already been fixed in SDL Mixer, there just hasn't been a new release yet.

**Comment by QuartzSTQ** [comment]
I've just tested and this is (obviously) present in Chocolate Doom's autobuild as well.

This should be fixed for the autobuilds at least but I guess they haven't updated SDL Mixer.

**Comment by fabiangreffrath** [comment]
We can submit a patch to MSYS2, sure, but is there a way we somehow can contribute this fix to the SDL2_Mixer package in vcpkg?

**Comment by QuartzSTQ** [comment]
The fix hasn't actually been released though, and I don't know whether vcpkg or MSYS2 use the main branches directly. Even then the version of SDL Mixer in the autobuilds is an outdated release (2.6.2 compared to 2.6.3) anyway.

**Comment by rfomin** [comment]
> We can submit a patch to MSYS2, sure, but is there a way we somehow can contribute this fix to the SDL2_Mixer package in vcpkg?

We can submit patch to vcpkg the same way, or just build it ourselves. But we don't use vcpkg for Crispy/Choco.

**Comment by fabiangreffrath** [comment]
> We can submit patch to vcpkg the same way, or just build it ourselves. But we don't use vcpkg for Crispy/Choco.

But isn't SDL2_Mixer in msys2 using libflac and the bug in question only occurs with the built-in drflac?

**Comment by rfomin** [comment]
> But isn't SDL2_Mixer in msys2 using libflac and the bug in question only occurs with the built-in drflac?

You're right, drflac seems to have a higher priority. I haven't investigated this.

**Comment by mikeday0** [comment]
> 
> You're right, drflac seems to have a higher priority. I haven't investigated this.

`--disable-music-flac-drflac` probably needs to be passed to `configure`.

**Comment by fabiangreffrath** [comment]
Bummer, no trace of libflac:

```
 $ objdump -x /mingw32/bin/SDL2_mixer.dll | grep "DLL Name"
        DLL Name: libgcc_s_dw2-1.dll
        DLL Name: KERNEL32.dll
        DLL Name: libmpg123-0.dll
        DLL Name: msvcrt.dll
        DLL Name: libopusfile-0.dll
        DLL Name: SDL2.dll
        DLL Name: WINMM.DLL
```

**Comment by mikeday0** [comment]
https://github.com/chocolate-doom/chocpkg/pull/24 will address this problem for the autobuilds.

**Comment by LaggyAsBalls** [comment]
this bug is still present in the latest builds. oddly, it doesn't seem to be present in the older build i just happen to have on my pc.

**Comment by mikeday0** [comment]
> this bug is still present in the latest builds. oddly, it doesn't seem to be present in the older build i just happen to have on my pc.

Looks like this FLAC issue might still be present on the latest.chocolate-doom.org builds. Instead, please try the Github Actions artifacts. ([This](https://github.com/fabiangreffrath/crispy-doom/actions/runs/29360098462) is the latest one at the time of this posting.)
